cvmfs/catalog.py

```diff
@@ -209,10 +209,14 @@ def find_nested_for_path(self, needle_path):
         nested_catalogs  = self.list_nested()
         best_match       = None
         best_match_score = 0
+        num_slashes      = needle_path.count('/')
```

Comment:
probably you want to retrieve this information from `real_needle_path` two lines below.